addons/mail/static/src/components/discuss/discuss_tests.js

```diff
@@ -421,6 +421,175 @@ QUnit.test('sidebar: add channel', async function (assert) {
+                id: 20,
+                name: "General",
+            }],
+        },
+    });
+    await this.start({});
+    assert.containsOnce(
+        document.body,
+        '.o_Discuss_thread[data-thread-model="mail.box"][data-thread-id="inbox"]',
+        "The inbox is displayed"
+    );
+    assert.containsOnce(document.body, '.o_DiscussSidebar_list .o_DiscussSidebarItem', "1 channel is present");
+    assert.strictEqual(
+        document.querySelector(`.o_DiscussSidebar_list .o_DiscussSidebarItem .o_DiscussSidebarItem_name`).textContent,
+        'General',
+        "The channel name is 'General'"
+    );
+});
+
+QUnit.test('sidebar: pinned channel 2: open pinned channel', async function (assert) {
+    assert.expect(1);
+
+    Object.assign(this.data.initMessaging, {
+        channel_slots: {
+            channel_channel: [{
+                channel_type: "channel",
+                id: 20,
+                name: "General",
+            }],
+        },
+    });
+    await this.start({});
+    await afterNextRender(() =>
+        document.querySelector('.o_DiscussSidebar_list .o_DiscussSidebarItem').click()
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_Discuss_thread[data-thread-model="mail.channel"][data-thread-id="20"]',
+        "The channel #General is displayed"
+    );
+});
+
+QUnit.test('sidebar: pinned channel 3: open pinned channel and unpin it', async function (assert) {
+    assert.expect(8);
+
+    Object.assign(this.data.initMessaging, {
+        channel_slots: {
+            channel_channel: [{
+                channel_type: "channel",
+                id: 20,
+                name: "General",
+            }],
+        },
+    });
+    let env;
+    await this.start({
+        async mockRPC(route, args) {
+            if (args.method === 'execute_command') {
+                assert.step('execute_command');
+                assert.deepEqual(args.args[0], [20],
+                    "The right id is sent to the server to remove"
+                );
+                assert.strictEqual(args.args[1], 'leave',
+                    "The right command is sent to the server"
+                );
+                const notifConfirmUnpin = [
+                    ["dbName", 'res.partner', 1],
+                    {id: 20, name: "General", state: 'open', channel_type: 'channel', public: 'public', info: 'unsubscribe'}
+                ];
+                env.services.bus_service.trigger('notification', [notifConfirmUnpin]);
+                return;
+            }
+            if (args.method === 'channel_fold') {
+                assert.step('channel_fold');
+                const notifConfirmFold = [
+                    ["dbName", 'res.partner', 1],
+                    {id: 20, name: "General", state: 'closed', is_minimized: false, channel_type: 'channel', public: 'public'}
+                ];
+                env.services.bus_service.trigger('notification', [notifConfirmFold]);
+            }
```

Comment:
I would prefer updating `channel_fold` management in `mail/mock_server.js`, so that it uses `this.data` accordingly. Right now It's defined just to avoid `unimplemented route` crash when we don't bother with it, but the correct way is to fully manage using `data`.

I'm good with test just adding a mockRPC for the `assert.step()`, and then just using mock server behaviour.